Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -393,7 +393,11 @@ public static Branch getInstance(Context context, String branchKey) {
 		}
 		branchReferral_.context_ = context.getApplicationContext();
 		if (branchKey.startsWith("key_")) {
-			branchReferral_.prefHelper_.setBranchKey(branchKey);
+			boolean isNewBranchKeySet = branchReferral_.prefHelper_.setBranchKey(branchKey);
+			//on setting a new key clear link cache
+			if (isNewBranchKeySet) {
+				branchReferral_.linkCache_.clear();
```

Comment:
See the [iOS SDK implementation](https://github.com/BranchMetrics/Branch-iOS-Deep-Linking-SDK/blob/aa0fa04364ddbc5abb19461e01e1f93b5e48e84e/Branch-SDK/Branch-SDK/Branch.m#L829)